### DISCUSSION TITLE
Zustand not working with Next.js (SSR)

### DISCUSSION THREAD

**Discussion by gabrielgrs** [pmndrs/zustand#825] [Q&A]
Hello people!

I'm trying to use Zustand inside a Next application, I've configured all the rehydration part, but the state is always getting lost;

I tried to use the example from the Next Github (https://stackblitz.com/github/vercel/next.js/tree/canary/examples/with-zustand), but even so, when switching screens, the state is getting lost

Can anyone guide me on how to use the two together?

Thank you!

**Comment by dai-shi** [comment]
Can anyone help? cc @Munawwar

**Comment by Munawwar** [comment]
Hi @gabrielgrs.
Which state is getting lost?
On SSR and SSG pages it's coded in your store.js (in useLayoutEffect) to replace states with whatever the server sent back, if that is what you meant by states getting lost. It is also the correct behavior - it is like a page refresh (Assume server takes over those pages for first render)
